File: articles/al-kindi-frequency-analysis-and-the-birth-of-cryptanalysis/typescript/frequency_count.ts
const text = "THIS IS A TOY CLASSICAL CIPHER EXAMPLE";
const counts: Record<string, number> = {};
for (const ch of text.toLowerCase()) {
  if (/[a-z]/.test(ch)) counts[ch] = (counts[ch] ?? 0) + 1;
}
console.log(counts);

File: articles/al-kindi-frequency-analysis-and-the-birth-of-cryptanalysis/java/FrequencyCount.java
import java.util.Map;
import java.util.TreeMap;

public class FrequencyCount {
    public static void main(String[] args) {
        String text = "THIS IS A TOY CLASSICAL CIPHER EXAMPLE";
        Map<Character, Integer> counts = new TreeMap<>();
        for (char ch : text.toLowerCase().toCharArray()) {
            if (Character.isLetter(ch)) {
                counts.put(ch, counts.getOrDefault(ch, 0) + 1);
            }
        }
        System.out.println(counts);
    }
}
